api/RS.API/RS.Service/Logic/CandidateManagerService.cs

```diff
@@ -4,19 +4,171 @@
+        public CandidateManagerService(IPrincipal principal, ICandidateRepository candidateRepository, IOpeningRepository openingRepository, IQualificationRepository qualificationRepository)
         {
+            _principal = principal as ClaimsPrincipal;
             _candidateRepository = candidateRepository;
+            _openingRepository = openingRepository;
+            _qualificationRepository = qualificationRepository;
         }
 
+        public IResult AddCandidate(CandidateViewModel candidate)
+        {
+            var result = new Result
+            {
+                Operation = Operation.Create,
+                Status = Status.Success
+            };
+            try
+            {
+                var candidateModel = new Candidates();
+                candidateModel.MapFromViewModel(candidate, (ClaimsIdentity)_principal.Identity);
+                candidateModel.QualificationId = candidate.Qualification;
+                candidateModel.Qualification = _qualificationRepository.GetByID(candidate.Qualification);
+
+                var openingCandidate = new OpeningCandidates();
+                openingCandidate.OpeningId = candidate.Opening;
+                openingCandidate.candidate = candidateModel;
+                openingCandidate.Opening = _openingRepository.GetByID(candidate.Opening);
+                openingCandidate.MapAuditColumns((ClaimsIdentity)_principal.Identity);
+
+                var organization = new Organizations();
+                organization.Name = candidate.Organization;
+                organization.MapAuditColumns((ClaimsIdentity)_principal.Identity);
+
+                _candidateRepository.AddCandidate(candidateModel, openingCandidate, organization);
+                _candidateRepository.SaveChanges();
+                result.Body = candidateModel;
+            }
+            catch (Exception e)
+            {
+                result.Message = e.Message;
+                result.Status = Status.Error;
+            }
+            return result;
+        }
 
+        public IResult DeleteCandidate(Guid id)
+        {
+            throw new NotImplementedException();
+        }
+
+        public IResult GetAllCandidate()
+        {
+            var result = new Result
+            {
+                Operation = Operation.Read,
+                Status = Status.Success
+            };
+            try
+            {
+                var allCandidates = _candidateRepository.GetAll();
+                var candidateList = allCandidates.Select(candidate =>
+                {
+                    var openingCandidate = _candidateRepository.GetOpeningCandidate(candidate.CandidateId);
+                    var candidateViewModel = new CandidateListModel();
+                    if (openingCandidate != null)
+                    {
+                        candidateViewModel.Opening = openingCandidate.Opening.Title;
+                        candidateViewModel.ModifiedDate = openingCandidate.Opening.ModifiedDate;
```

Comment:
rename properties correspond to there model names

Reply:
Done